### PR TITLE
Build a experimental flannel image with the kube-subnet-manager PR included

### PR DESCRIPTION
**Pull request by luxas**
@mikedanese Can you test and push this?
I'm also testing right now...

cc @kubernetes/sig-cluster-lifecycle 
Here's the flannel daemonset config and a Makefile for building and pushing it.

It would be great if users could choose between three (or four) options: weave, flannel, calico and canal!

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33115)

### CODE REVIEW

**Comment by luxas** [comment]
@k8s-bot gce e2e test this

File: cluster/images/flannel/Makefile [hunk 1]
```diff
@@ -15,52 +15,31 @@
 # Build the flannel image
```
Comment (mikedanese) [inline cluster/images/flannel/Makefile]: Why are we even maintaining this? This seems something the flannel guys should be publishing.

**Comment by mikedanese** [comment]
LGTM, since it's already there but i think we should just rm -rf this directory. We have so much cruft in our codebase.

File: cluster/images/flannel/flannel-ds.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            fieldRef:
+              fieldPath: metadata.namespace
+        volumeMounts:
+        - name: run
+          mountPath: /run
+        - name: flannel-cfg
+          mountPath: /etc/kube-flannel/
+      - name: install-cni
+        image: gcr.io/google_containers/flannel-amd64:0.6.2-kube
+        command: [ "/bin/sh", "-c", "set -e -x; TMP=/cni/net.d/.tmp-flannel-cfg; cp /etc/kube-flannel/cni-conf.json ${TMP}; mv ${TMP} /cni/net.d/10-flannel.conf; while true; do sleep 3600; done" ]
```
Comment (tomdee) [inline cluster/images/flannel/flannel-ds.yaml]: Shouldn't TMP=/cni/net.d... be TMP=/etc/cni/net.d...

**Comment by errordeveloper** [comment]
Didn't we declare a feature freeze on `/cluster`?

File: cluster/images/flannel/flannel-ds.yaml (hunk 2 above)
Comment (luxas) [inline cluster/images/flannel/flannel-ds.yaml]: Good catch!
Just copied this from flannel/Documentation...
Should be fixed there as well

**Comment by luxas** [comment]
@errordeveloper This isn't the same as the deprecation of kube-up features.
This is about making the flannel daemonset available for kubeadm use, although flannel hasn't released a version with the patch yet

@mikedanese If flannel releases a `v0.6.2` with your patch **now**, I'm going to `rm -rf` this dir.
@tomdee What do you think about that?

Otherwise, @mikedanese should just build these images and push after we've verified it's working once again.

This directory wasn't unnecessary when I made it; it built flannel `v0.5.5` images (which has been used a lot) for all arches before flannel multiarch was merged in `v0.6.0`

**Comment by tomdee** [comment]
@mikedanese @luxas I did a flannel v0.6.2 release yesterday but the kube flannel feature was _not_ in it.  Sorry. However, I've improved the flannel-git builds so there's now a tag for each master commit so you could just use one of those as your release? https://quay.io/repository/coreos/flannel-git?tab=tags

**Comment by luxas** [comment]
Seems like flannel-git suits our use case.
I'll close this PR and remove this code from Kubernetes if it works

Thanks @tomdee for the releasing work, it helps a lot!

**Pull request closed without merging**